### ISSUE TITLE
GraphQL Flat render does not resolve related items

### ISSUE THREAD

**Issue by AreeseDT** [VirtusLab-Open-Source/strapi-plugin-navigation#201]
The `renderNavigation` query is not resolving related content items when the NavigationRenderType is FLAT. This causes any navigation item that has related content to be returned as null.

Error returned:
```
{
  "message": "Abstract type \"NavigationRelated\" must resolve to an Object type at runtime for field \"NavigationItem.related\". Either the \"NavigationRelated\" type should provide a \"resolveType\" function or each possible type should provide an \"isTypeOf\" function.",
  "locations": [
    {
      "line": 27,
      "column": 3
    }
  ],
  "path": [
    "renderNavigation",
    4,
    "related"
  ],
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR",
    "exception": {
      "message": "Abstract type \"NavigationRelated\" must resolve to an Object type at runtime for field \"NavigationItem.related\". Either the \"NavigationRelated\" type should provide a \"resolveType\" function or each possible type should provide an \"isTypeOf\" function.",
      "locations": [
        {
          "line": 27,
          "column": 3
        }
      ],
      "stacktrace": [
        "GraphQLError: Abstract type \"NavigationRelated\" must resolve to an Object type at runtime for field \"NavigationItem.related\". Either the \"NavigationRelated\" type should provide a \"resolveType\" function or each possible type should provide an \"isTypeOf\" function.",
        "    at ensureValidRuntimeType (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\graphql\\execution\\execute.js:685:11)",
        "    at completeAbstractValue (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\graphql\\execution\\execute.js:680:42)",
        "    at completeValue (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\graphql\\execution\\execute.js:584:12)",
        "    at D:\\Projects\\Sandbox\\plugin-test\\node_modules\\graphql\\execution\\execute.js:469:16",
        "    at runMicrotasks (<anonymous>)",
        "    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
        "    at async Promise.all (index 10)",
        "    at async Promise.all (index 4)",
        "    at async Promise.all (index 0)",
        "    at async execute (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\apollo-server-core\\dist\\requestPipeline.js:205:20)",
        "    at async Object.processGraphQLRequest (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\apollo-server-core\\dist\\requestPipeline.js:139:28)",
        "    at async processHTTPRequest (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\apollo-server-core\\dist\\runHttpQuery.js:179:30)",
        "    at async D:\\Projects\\Sandbox\\plugin-test\\node_modules\\apollo-server-koa\\dist\\ApolloServer.js:82:59",
        "    at async bodyParser (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\koa-bodyparser\\index.js:95:5)",
        "    at async returnBodyMiddleware (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\@strapi\\strapi\\lib\\services\\server\\compose-endpoint.js:52:18)",
        "    at async policiesMiddleware (D:\\Projects\\Sandbox\\plugin-test\\node_modules\\@strapi\\strapi\\lib\\services\\server\\policy.js:24:5)"
      ]
    }
  }
}
```

**Comment by cyp3rius** [comment]
@AreeseDT have you configured plugin using `Settings -> Navigation plugin -> Configuration` ?

I just did test on the most recent version and works as below:

```gql
query {
  renderNavigation(
    navigationIdOrSlug: "main-navigation"
    type: TREE
    menuOnly: false
  ) {
    id
    title
    path
    related {
      __typename

      ... on Page {
        Title
      }

      ... on WithFlowType {
        Name
      }
    }
    items {
      id
      title
      path
      related {
        __typename

        ... on Page {
          Title
        }

        ... on WithFlowType {
          Name
        }
      }
    }
  }
}
```

```json
{
  "data": {
    "renderNavigation": [
      {
        "id": 8,
        "title": "Test page",
        "path": "/test-path",
        "related": {
          "__typename": "WithFlowType",
          "Name": "Test"
        },
        "items": [
          {
            "id": 11,
            "title": "Nested",
            "path": "/test-path/nested-one",
            "related": {
              "__typename": "Page",
              "Title": "ghghghgh"
            }
          }
        ]
      },
      {
        "id": 10,
        "title": "Another page",
        "path": "/another",
        "related": {
          "__typename": "Page",
          "Title": "dfdfdf"
        },
        "items": []
      }
    ]
  }
}
```

I'm working on Readme.md updated to make it more concrete.

**Comment by AreeseDT** [comment]
The issue only occurs when `type` is `FLAT`.

```
query {
  renderNavigation(
    navigationIdOrSlug: "main-navigation"
    type: FLAT
    menuOnly: false
  ) {
    ...
  }
}
```

**Comment by cyp3rius** [comment]
Fix released in version `2.0.13`
